sdk/src/azure/core/az_http_request.c

```diff
@@ -35,8 +30,15 @@ AZ_NODISCARD az_result az_http_request_init(
   _az_PRECONDITION_VALID_SPAN(headers_buffer, 0, false);
 
   int32_t query_start = 0;
-  az_result url_with_query = _az_span_scan_until(
-      az_span_create(az_span_ptr(url), url_length), _az_is_question_mark, &query_start);
+  uint8_t* ptr = az_span_ptr(url);
+  for (; query_start < url_length; ++query_start)
```

Comment:
if ptr is only used inside the for() loop, why don't you add it as part of the for init? like
for(uint8_t* ptr = az_span_ptr(url); query_start < .....) {   ... }

Also, I remember we were trying to avoid using `az_span_ptr()` ??   Is there a way we could avoid it here?

lastly, maybe make ptr to be const?   `uint8_t const * const ptr = az_span_ptr(...)`  then we make sure it's not mutated?

Reply:
> if ptr is only used inside the for() loop, why don't you add it as part of the for init? like
for(uint8_t* ptr = az_span_ptr(url); query_start < .....) {   ... }

I don't like the readability of that pattern, where we aren't following the standard `for (start; bounds condition; increment)`. Declaring the ptr inside the for loop body makes it seem like it will change within the loop body, when it doesn't.

> Also, I remember we were trying to avoid using `az_span_ptr()` ??   Is there a way we could avoid it here?

We need to access the byte content of the span. The only way to do that is with az_span_ptr. If we had a way to define an indexer operator in C, that problem would go away.

> lastly, maybe make ptr to be const?   `uint8_t const * const ptr = az_span_ptr(...)`  then we make sure it's not mutated?

Good idea.